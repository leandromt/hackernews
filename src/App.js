import React, { Component } from 'react';

const list = [
  {
    title: 'React',
    url: 'https://facebook.github.io/react/',
    author: 'Jordan Walke',
    num_comments: 3,
    points: 4,
    objectID: 0,
  },
  {
    title: 'Redux',
    url: 'https://github.com/reactjs/redux',
    author: 'Dan Abramov, Andrew Clark',
    num_comments: 2,
    points: 5,
    objectID: 1,
  },
];

class App extends Component {
  render() {
    return (
      <div className="App">
        <ul>
          {
            list.map(function (item, key) {
              return (
                <li id={item.key} key={item.objectID}>
                  <p><a href={item.ul}>{item.title}</a></p>
                  <p>{item.author}</p>
                  <p>{item.num_comments}</p>
                  <p>{item.points}</p>
                  <p>{item.objectID}</p>
                </li>
              );
            })
          }
        </ul>
      </div>
    );
  }
}

export default App;
