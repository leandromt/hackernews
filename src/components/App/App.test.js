import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "./App";
import { Button } from "./../Buttons";

describe("App", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test("has a valid snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Button", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test("has a valid snapshot", () => {
    const component = renderer.create(<Button>More</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
