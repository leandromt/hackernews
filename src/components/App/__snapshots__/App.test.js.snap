// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <div>
      <h1>
        Hacker News API
      </h1>
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        id="search"
        onChange={[Function]}
        onSubmit={[Function]}
        type="text"
        value="react"
      />
      <button
        type="submit"
      >
        Buscar
      </button>
    </form>
  </div>
  <div
    className="table"
  />
  <div
    className="interactions"
  >
    <button
      className="btn-default"
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`Button has a valid snapshot 1`] = `
<button
  className="btn-default"
  onClick={undefined}
  type="button"
>
  More
</button>
`;
